io/disk/ltp_fsstress.py

```diff
@@ -60,6 +60,13 @@ def setUp(self):
         self.p_val = self.params.get('p_val', default='100')
 
         smm = SoftwareManager()
+        #if self.dir is None using using self.workdir as self.dir
+        if self.disk:
+            if not self.dir:
+                self.dir = self.workdir
```

Comment:
@AYUSHJAIN951 if the disk is not given then also test will pass here. kind of agree with it.
But if disk is  given but it is not available in OS (could be some typo or having some partition on it etc.) or issue in parsing the yaml parameter to the test (we have seen this in the past) then also test will be running on the local disk and it will pass after above changes.   and the end user will be under impression that test is ran on given disk but not in real. And the whole intention of running the test on some block device will not be fulfill here.

Second reason is test like ltp, fio creates test temp file on the given dir, so one should really aware of disk space available in the given dir.

 Best way to cover this scenario will be using the another yaml file and hard code the /mnt or use extra parameter in the yaml file like `default_dir` and handle the same in the .py code like below,

```
if default_dir:
    self.dir = default_dir
```
hope this will help your scenario. tuning yaml is easiest way than changing the code flow as it will have major impact :-)

Reply:
@Naresh-ibm First the case of Disk being misspelled and similar cases ends up with a "Missing Disk in OS" prompt itself with this solution and doesn't run the test ahead.

Reply:
Secondly previously in your comment you referred that "/mnt" can't be used and in your recent comment you are referring to use "/mnt" default its getting quite confusing for me personally to understand, so could you please explain it more in detail.
And let's collaborate together and get this to a better solution.

Reply:
@AYUSHJAIN951 
1. if test ends with "Missing Disk in OS", then your disk is not available in the OS or it has some partition already hence it is not found in json format. please check the logic in the Library for this. if you still did not understand it, let me know I get you the exact logic.
2. I asked you earlier,  not to hard code /mnt in the .py file as a default parameter. Now I am suggesting you to us this parameter in a separate yaml file which will not be affecting the .py file(As existing .py not helping you for your test  scenario). In the new yaml file you can hard code all your dir values whichever you want. OR in other scenarios, where you want to use the current for test case, then for this case to handle,  I suggested to add a new parameter in yaml called, `default_dir` and add the related logic in the .py file as suggest above. this way it will not be effecting anybody and actual test scenario meant for block devices.